src/pcs/c1/PageMods.js

```diff
@@ -0,0 +1,145 @@
+import BodySpacingTransform from '../../transform/BodySpacingTransform'
+import CollapseTable from '../../transform/CollapseTable'
+import DimImagesTransform from '../../transform/DimImagesTransform'
+import LazyLoadTransformer from '../../transform/LazyLoadTransformer'
+import PlatformTransform from '../../transform/PlatformTransform'
+import Scroller from './Scroller'
+import ThemeTransform from '../../transform/ThemeTransform'
+
+/**
+ * Executes common JS functionality the client should start, like hooking up events for
+ * lazy loading and table collapsing/expanding.
+ * Client side complement of server-side DOM transformations.
+ * Note: this will be called automatically when DOM is ready
+ * @param {!Window} window
+ * @param {!Document} document
+ * @return {void}
+ */
+const onPageLoad = (window, document) => {
+  const lazyLoader = new LazyLoadTransformer(window, 2)
+  lazyLoader.collectExistingPlaceholders(document.body)
+  lazyLoader.loadPlaceholders()
+
+  CollapseTable.setupEventHandling(window, document, true, Scroller.scrollWithDecorOffset)
+}
+
+/**
+ * @typedef {function} OnSuccess
+ * @return {void}
+ */
+
+/**
+ * Makes multiple page modifications based on client specific settings, which should be called
+ * during initial page load.
+ * @param {!Document} document
+ * @param {!{}} settings client settings
+ *   { platform, clientVersion, theme, dimImages, margins, areTablesCollapsed, scrollTop }
+ * @param {?OnSuccess} onSuccess callback
+ * @return {void}
+ */
+const setMulti = (document, settings, onSuccess) => {
+  if (settings.platform !== undefined) {
+    PlatformTransform.setPlatform(document, settings.platform)
+  }
+  if (settings.theme !== undefined) {
+    ThemeTransform.setTheme(document, settings.theme)
+  }
+  if (settings.dimImages !== undefined) {
+    DimImagesTransform.dimImages(document, settings.dimImages)
+  }
+  if (settings.margins !== undefined) {
+    BodySpacingTransform.setMargins(document.body, settings.margins)
+  }
+  if (settings.areTablesCollapsed) {
+    CollapseTable.toggleCollapsedForAll(document.body)
+  }
+  if (settings.scrollTop !== undefined) {
+    Scroller.setScrollTop(settings.scrollTop)
+  }
+
+  if (onSuccess instanceof Function) {
+    onSuccess()
+  }
+}
+
+/**
+ * Sets the theme.
+ * @param {!Document} document
+ * @param {!string} theme one of the values in Themes
+ * @param {?OnSuccess} onSuccess callback
+ * @return {void}
+ */
+const setTheme = (document, theme, onSuccess) => {
+  ThemeTransform.setTheme(document, theme)
+
+  if (onSuccess instanceof Function) {
+    onSuccess()
+  }
+}
+
+/**
+ * Turns on dimming of images.
```

Comment:
More accurate to say "toggles dimming of images" since this can also turn it off.